library/ssl_msg.c

```diff
@@ -1156,33 +1131,8 @@ int mbedtls_ssl_decrypt_buf( mbedtls_ssl_context const *ssl,
     if( mode == MBEDTLS_MODE_STREAM )
 #endif /* MBEDTLS_USE_PSA_CRYPTO */
     {
-        padlen = 0;
-
-#if defined(MBEDTLS_USE_PSA_CRYPTO)
-    /* The only stream "cipher" we support is "NULL" */
-    if ( transform->psa_alg != MBEDTLS_SSL_NULL_CIPHER )
-        return( MBEDTLS_ERR_SSL_INTERNAL_ERROR );
-
-    olen = rec->data_len;
-#else
-
-        if( ( ret = mbedtls_cipher_crypt( &transform->cipher_ctx_dec,
-                                   transform->iv_dec,
-                                   transform->ivlen,
-                                   data, rec->data_len,
-                                   data, &olen ) ) != 0 )
-        {
-            MBEDTLS_SSL_DEBUG_RET( 1, "mbedtls_cipher_crypt", ret );
-            return( ret );
-        }
-#endif /* MBEDTLS_USE_PSA_CRYPTO */
-
-        if( rec->data_len != olen )
-        {
-            MBEDTLS_SSL_DEBUG_MSG( 1, ( "should never happen" ) );
-            return( MBEDTLS_ERR_SSL_INTERNAL_ERROR );
-        }
-
+        /* The only supported stream cipher is "NULL",
+         * so there's nothing to do here.*/
```

Comment:
Note: the block used to set `padlen` and `olen`, which have a wider scope than just this block (while in the encrypt case, `olen` was local to the block that we're simplifying). In particular, I think `padlen` will be used outside the block, because we need it to locate the HMAC that we're checking later.

But honestly, this function is pretty hard to follow, with multiple case interlaced (and grouped differently in different places), so let's wait and see what the CI says. If it passes, fine, otherwise I think adding back `padlen = 0;` should be the first thing we try.

Reply:
OK, the CI's happy so I'm happy as well. (I mean, I'm happy with this PR - I'm not happy with the function being hard to follow, but that's a long-standing issue.)

Reply:
I have built locally with and without psa crypto, so should be fine.